Two tracebacks with assert_wrapper / wrap_assert_has_calls

When using `assert_has_calls` I am seeing two tracebacks, but the first one should not be there?!

```
Traceback (most recent call last):
  File "…/pytest-mock/pytest_mock.py", line 155, in assert_wrapper
    __wrapped_mock_method__(*args, **kwargs)
  File "/usr/lib64/python3.5/unittest/mock.py", line 824, in assert_has_calls
    ) from cause
AssertionError: Calls not found.
Expected: […]
Actual: […]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "…/app/tests/test_metrics.py", line 495, in test_handle_missing_values
    expected_call(end, 2, None),
  File "…/pytest-mock/pytest_mock.py", line 181, in wrap_assert_has_calls
    *args, **kwargs)
  File "…/pytest-mock/pytest_mock.py", line 157, in assert_wrapper
    raise AssertionError(*e.args)
AssertionError: Calls not found.
Expected: […]
Actual: […]
```

This is with pytest-2.9.2.dev1 and Python 3.5.1.

It looks like the `__tracebackhide__ = True` trick is not working?!

With `mock_traceback_monkeypatch=false` I only get the original traceback as expected.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -14,10 +14,16 @@
   ``repr`` string as well as related assertion failure messages.
   Thanks `@jurko-gospodnetic`_ for the PR (`#40`_).
 
+* Monkey patching is automatically disabled with the ``--tb=native`` option. The underlying
+  mechanism used to suppress traceback entries from ``mock`` module does not work with that option
+  anyway plus it generates confusing messages on Python 3.5 due to exception chaining (`#44`_).
+  Thanks `@blueyed`_ for the report.
+
 .. _@jurko-gospodnetic: https://github.com/jurko-gospodnetic
 .. _@asfaltboy: https://github.com/asfaltboy
-.. _#40: https://github.com/pytest-dev/pytest-mock/issues/40
 .. _#36: https://github.com/pytest-dev/pytest-mock/issues/36
+.. _#40: https://github.com/pytest-dev/pytest-mock/issues/40
+.. _#44: https://github.com/pytest-dev/pytest-mock/issues/44
 
 1.0
 ---

--- a/pytest_mock.py
+++ b/pytest_mock.py
@@ -252,5 +252,6 @@ def parse_ini_boolean(value):
 
 
 def pytest_configure(config):
-    if parse_ini_boolean(config.getini('mock_traceback_monkeypatch')):
+    tb = config.getoption('--tb')
+    if parse_ini_boolean(config.getini('mock_traceback_monkeypatch')) and tb != 'native':
         wrap_assert_methods(config)

--- a/test_pytest_mock.py
+++ b/test_pytest_mock.py
@@ -370,7 +370,7 @@ def assert_argument_introspection(left, right):
         raise AssertionError("DID NOT RAISE")
 
 
-@pytest.mark.skipif(sys.version_info[0] == 3 and sys.version_info[1] in (3, 4),
+@pytest.mark.skipif(sys.version_info[:2] in [(3, 3), (3, 4)],
                     reason="assert_not_called not available in python 3.3 and 3.4")
 def test_assert_not_called_wrapper(mocker):
     stub = mocker.stub()
@@ -489,3 +489,21 @@ def test_patched_method_parameter_name(mocker):
     m = mocker.patch.object(Request, 'request')
     Request.request(method='get', args={'type': 'application/json'})
     m.assert_called_once_with(method='get', args={'type': 'application/json'})
+
+
+def test_monkeypatch_native(testdir):
+    testdir.makepyfile("""
+        def test_foo(mocker):
+            stub = mocker.stub()
+            stub(1, greet='hello')
+            stub.assert_called_once_with(1, greet='hey')
+    """)
+    if hasattr(testdir, 'runpytest_subprocess'):
+        result = testdir.runpytest_subprocess('--tb=native')
+    else:
+        # pytest 2.7.X
+        result = testdir.runpytest('--tb=native')
+    assert result.ret == 1
+    assert 'Traceback (most recent call last)' in result.stdout.str()
+    assert 'During handling of the above exception' not in result.stdout.str()
+    assert 'Differing items:' not in result.stdout.str()

